### DISCUSSION TITLE
Build>Core>Print build failure

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-Ed/Community#281] [Installation] [answered]
Hi Team,

CSP: GCP
Sunbirdevops- 5.1.0
I am trying to build a print it fails
Build version -release-3.8.0_RC1

`Started by user Prasath Sivasubramaniyan
Checking out git https://github.com/project-sunbird/print-service.git into /var/lib/jenkins/workspace/Build/Core/Print@script/e231b1aaba5cdb9952a5aec15b9f4c6379b3c0c768072339b06694602c0a5470 to read Jenkinsfile
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/project-sunbird/print-service.git
 > git init /var/lib/jenkins/workspace/Build/Core/Print@script/e231b1aaba5cdb9952a5aec15b9f4c6379b3c0c768072339b06694602c0a5470 # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/print-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/print-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/print-service.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-3.8.0_RC1^{commit} # timeout=10
 > git rev-parse release-3.8.0_RC1^{commit} # timeout=10
Checking out Revision 92807caa9b0ceb02087649ce01f369134f001917 (release-3.8.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 92807caa9b0ceb02087649ce01f369134f001917 # timeout=10
Commit message: "Merge pull request #22 from keshavprasadms/release-3.8.0"
First time build. Skipping changelog.
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/workspace/Build/Core/Print
[Pipeline] {
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (Checkout)
[Pipeline] echo
[1m[32mFound environment variable named hub_org with value as: asia.gcr.io/upsmf-368011[0m
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/project-sunbird/print-service.git
 > git init /var/lib/jenkins/workspace/Build/Core/Print # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/print-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/print-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/print-service.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-3.8.0_RC1^{commit} # timeout=10
 > git rev-parse release-3.8.0_RC1^{commit} # timeout=10
Checking out Revision 92807caa9b0ceb02087649ce01f369134f001917 (release-3.8.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 92807caa9b0ceb02087649ce01f369134f001917 # timeout=10
Commit message: "Merge pull request #22 from keshavprasadms/release-3.8.0"
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Build/Core/Print/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/print-service.git # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/print-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/print-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse refs/remotes/origin/refs/tags/release-3.8.0_RC1^{commit} # timeout=10
 > git rev-parse refs/tags/release-3.8.0_RC1^{commit} # timeout=10
Checking out Revision 92807caa9b0ceb02087649ce01f369134f001917 (refs/tags/release-3.8.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 92807caa9b0ceb02087649ce01f369134f001917 # timeout=10
Commit message: "Merge pull request #22 from keshavprasadms/release-3.8.0"
[Pipeline] echo
[1m[33mgithub_release_tag specified, building from github_release_tag: release-3.8.0_RC1[0m
[Pipeline] echo
build_tag: release-3.8.0_RC1
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] echo
Environment will be : build
[Pipeline] sh
+ chmod 777 build.sh
[Pipeline] sh
+ ./build.sh release-3.8.0_RC1 built-in asia.gcr.io/upsmf-368011
Sending build context to Docker daemon  174.6kB

Step 1/18 : FROM circleci/node:8.11.2-stretch as build
 ---> 3c168695e85e
Step 2/18 : MAINTAINER "Mahesh Kumar Gangula" "mahesh@ilimi.in"
 ---> Running in 5425758bf73b
Removing intermediate container 5425758bf73b
 ---> cae5a83ea364
Step 3/18 : USER root
 ---> Running in 13437318c989
Removing intermediate container 13437318c989
 ---> c8e3f9f22e4e
Step 4/18 : COPY src /opt/print-service/
 ---> cb88388b6222
Step 5/18 : WORKDIR /opt/print-service/
 ---> Running in 0e9bd24e297a
Removing intermediate container 0e9bd24e297a
 ---> 6168b58682ce
Step 6/18 : RUN npm install --unsafe-perm
 ---> Running in 701a18751d84
[91mnpm WARN notice [SECURITY] log4js has the following vulnerability: 1 moderate. Go here for more details: https://github.com/advisories?query=log4js - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.
[0m[91mnpm WARN deprecated log4js@4.5.1: 4.x is no longer supported. Please upgrade to 6.x or higher.
[0m[91mnpm WARN deprecated puppeteer@2.0.0: Version no longer supported. Upgrade to @latest
[0m[91mnpm WARN deprecated uuid@3.2.1: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[0m[91mnpm WARN deprecated request@2.87.0: request has been deprecated, see https://github.com/request/request/issues/3142
[0m[91mnpm WARN deprecated[0m[91m superagent@5.3.1: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.
[0m[91mnpm WARN deprecated date-format@2.1.0: 2.x is no longer supported. Please upgrade to 4.x or higher.
[0m[91mnpm WARN deprecated streamroller@1.0.6: 1.x is no longer supported. Please upgrade to 3.x or higher.
[0m[91mnpm WARN deprecated har-validator@5.0.3: this library is no longer supported
[0m[91mnpm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[0m[91mnpm WARN deprecated azure-storage@2.10.7: Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
[0m[91mnpm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
[0m[91mnpm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
[0m[91mnpm WARN deprecated har-validator@5.1.5: this library is no longer supported
[0m
> puppeteer@2.0.0 install /opt/print-service/node_modules/puppeteer
> node install.js

[91m
[0mChromium downloaded to /opt/print-service/node_modules/puppeteer/.local-chromium/linux-706915
[91mnpm notice created a lockfile as package-lock.json. You should commit this file.
[0m[91mnpm WARN print-service@1.0.0 No description
[0m[91m
[0madded 208 packages in 19.039s
Removing intermediate container 701a18751d84
 ---> f4b37edf38ab
Step 7/18 : FROM node:8.11-slim
 ---> 1aa312dac871
Step 8/18 : MAINTAINER "Mahesh Kumar Gangula" "mahesh@ilimi.in"
 ---> Running in 2a457b9ebc10
Removing intermediate container 2a457b9ebc10
 ---> 840eacd560a6
Step 9/18 : RUN apt-get clean     && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -     && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list'     && apt update && apt install fonts-indic -y     && apt-get install -y google-chrome-unstable gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget       --no-install-recommends     && rm -rf /var/lib/apt/lists/*
 ---> Running in fd227df58192
OK
[91m
WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

[0mIgn http://deb.debian.org jessie InRelease
Get:1 http://security.debian.org jessie/updates InRelease [44.9 kB]
Get:2 http://deb.debian.org jessie-updates InRelease [16.3 kB]
Get:3 http://dl.google.com stable InRelease [1811 B]
Get:4 http://deb.debian.org jessie Release.gpg [1652 B]
Get:5 http://deb.debian.org jessie Release [77.3 kB]
Get:6 http://security.debian.org jessie/updates/main amd64 Packages [992 kB]
Ign http://deb.debian.org jessie-updates InRelease
Get:7 http://deb.debian.org jessie-updates/main amd64 Packages [20 B]
Ign http://deb.debian.org jessie Release
Get:8 http://dl.google.com stable/main amd64 Packages [1125 B]
Get:9 http://deb.debian.org jessie/main amd64 Packages [9098 kB]
Fetched 10.2 MB in 6s (1581 kB/s)
Reading package lists...
Building dependency tree...
Reading state information...
37 packages can be upgraded. Run 'apt list --upgradable' to see them.
[91mW: GPG error: http://deb.debian.org jessie-updates InRelease: The following signatures were invalid: KEYEXPIRED 1668891673
W: GPG error: http://deb.debian.org jessie Release: The following signatures were invalid: KEYEXPIRED 1668891673
[0m[91m
WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

[0mReading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  fonts-beng fonts-beng-extra fonts-deva fonts-deva-extra fonts-gargi
  fonts-gubbi fonts-gujr fonts-gujr-extra fonts-guru fonts-guru-extra
  fonts-kalapi fonts-knda fonts-lohit-beng-assamese fonts-lohit-beng-bengali
  fonts-lohit-deva fonts-lohit-gujr fonts-lohit-guru fonts-lohit-knda
  fonts-lohit-mlym fonts-lohit-orya fonts-lohit-taml
  fonts-lohit-taml-classical fonts-lohit-telu fonts-mlym fonts-nakula
  fonts-navilu fonts-orya fonts-orya-extra fonts-pagul fonts-sahadeva
  fonts-samyak-deva fonts-samyak-gujr fonts-samyak-mlym fonts-samyak-taml
  fonts-sarai fonts-smc fonts-taml fonts-telu fonts-telu-extra
The following NEW packages will be installed:
  fonts-beng fonts-beng-extra fonts-deva fonts-deva-extra fonts-gargi
  fonts-gubbi fonts-gujr fonts-gujr-extra fonts-guru fonts-guru-extra
  fonts-indic fonts-kalapi fonts-knda fonts-lohit-beng-assamese
  fonts-lohit-beng-bengali fonts-lohit-deva fonts-lohit-gujr fonts-lohit-guru
  fonts-lohit-knda fonts-lohit-mlym fonts-lohit-orya fonts-lohit-taml
  fonts-lohit-taml-classical fonts-lohit-telu fonts-mlym fonts-nakula
  fonts-navilu fonts-orya fonts-orya-extra fonts-pagul fonts-sahadeva
  fonts-samyak-deva fonts-samyak-gujr fonts-samyak-mlym fonts-samyak-taml
  fonts-sarai fonts-smc fonts-taml fonts-telu fonts-telu-extra
0 upgraded, 40 newly installed, 0 to remove and 37 not upgraded.
Need to get 3490 kB of archives.
After this operation, 12.3 MB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  fonts-deva-extra fonts-kalapi fonts-smc fonts-lohit-beng-bengali
  fonts-lohit-beng-assamese fonts-beng-extra fonts-beng fonts-lohit-deva
  fonts-sahadeva fonts-nakula fonts-samyak-deva fonts-deva fonts-gargi
  fonts-gubbi fonts-lohit-gujr fonts-samyak-gujr fonts-gujr-extra fonts-gujr
  fonts-lohit-guru fonts-guru-extra fonts-guru fonts-lohit-knda fonts-navilu
  fonts-knda fonts-lohit-mlym fonts-samyak-mlym fonts-mlym fonts-lohit-orya
  fonts-orya-extra fonts-orya fonts-pagul fonts-lohit-taml fonts-samyak-taml
  fonts-lohit-taml-classical fonts-taml fonts-lohit-telu fonts-telu-extra
  fonts-telu fonts-indic fonts-sarai
[91mE: There are problems and -y was used without --force-yes
[0mThe command '/bin/sh -c apt-get clean     && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -     && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list'     && apt update && apt install fonts-indic -y     && apt-get install -y google-chrome-unstable gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget       --no-install-recommends     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 100
Finished: FAILURE
`

**Comment by beepdot** [comment, marked as answer]
This service is not required. Please ignore the build and deploy of this service.

**Reply by tsprasath** [reply]
Close this thread
